src/java/org/apache/cassandra/service/snapshot/SnapshotLoader.java

```diff
@@ -80,7 +80,7 @@ public Set<TableSnapshot> loadSnapshots()
         {
             try
             {
-                if (dataDir.toFile().exists())
+                if (Files.exists(dataDir))
```

Comment:
When switching I always forget if there is a subtle difference with regard to bad disks...


`java.nio.file.Files#exists` does

```
} catch (IOException x) {
            // does not exist or unable to determine if file exists
            return false;
        }
```

and File calls `java.io.UnixFileSystem#getBooleanAttributes0`... which is native... it doesn't say it throws... so if disk is bad... does it fail or return false?

Reply:
`java.io.File` rarely throws, and doesn't in this case, so the two are consistent behaviourally. It even goes so far as to return 0 for most methods that return a long in the case of error. 

Or you could simply do `new File(dataDir).exists()`

Reply:
I'll just use `o.a.c.io.util.File#exists()` here and in `BinLog`.